src/Database/LSMTree/Internal/MergeSchedule.hs

```diff
@@ -872,7 +873,7 @@ mergeRuns resolve hfs hbio caching alloc runPaths mergeLevel runs = do
 supplyCredits ::
      (MonadSTM m, MonadST m, MonadMVar m, MonadMask m)
   => TableConfig
-  -> Credit
+  -> UnscaledCredits
   -> Levels m h
   -> m ()
 supplyCredits conf c levels =
```

Comment:
> calling it `UnscaledCredits` here and `Credits` in the `MergingRun` module feels like it is the wrong way around

I would argue calling it `ScaledCredits` in `MergingRun` feels like the wrong way around since the merging run itself doesn't have anything to do with scaling. For example when we use it inside the `MergingTree`, there is no scaling at all.

> maybe we should have separate `Credits` types with the same name in each module, and rely on qualified imports

That has has potential to be a little confusing, but it might be the best solution.

Reply:
(done)